### PR TITLE
Queues - Sofia Kim -api-muncher

### PR DESCRIPTION
**Pull request by Sofia15**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I went to their website, tried to figure what contents they have, then I moved to read the documentations. I used mainly postman to see what I could get through the api calls. |
| Describe your API Wrapper.  How did you decide on the methods you created? | I have two methods. One for to get recipes from Edamam site by search terms. The other one is to find the original recipes with uri. It was based on the requirements of the project that needs to make a request to the api using the search term and display the list to the user. Also, I needed a method that returns information about a specific recipe based on its ID.  |
| Describe an edge case or failure case test you wrote for your API Wrapper. | I have a test case to get no recipes for a bogus or blank search term.  |
| Explain how VCR aids in testing an API. | VCR is a tool that allows my app's tests to record HTTP interactions and replay them when necessary. By re-using these responses, VCR will improve stability and performance of my test suite. Each interaction is recorded as a cassette. We can load cassettes in out tests.  |
| What is the Heroku URL of your deployed application? | https://recipeworld-app.herokuapp.com |
| Do you have any recommendations on how we could improve this project for the next cohort? | It would be better if the requirements description was more clear or if we were gone over together with instructors. I made only limited 10 search items on my view page first because I misunderstood that it was what the requirement was. I couldn't understand the term "a paged list of recipes". After I  was going over my project at the mid point, I was told that it should be 10 items on multiple pages. |

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<body>
```
Comment (droberts-sea) [inline app/views/recipes/index.html.erb:1]: Individual view pages don't need a `<body>` tag, since that is included in `application.html.erb`.

**Comment by droberts-sea** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | yes |
| Comprehension questions	| yes |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | yes |
| Semantic HTML | yes |
| Errors are reported to the user | no - this should be one of the first things you do when setting up a new project, usually by setting up a section in `application.html.erb` to display messages from the flash. |
| API Wrapper to handle the API requests | The recipe details page didn't work for me, it threw an error every time. Looks like the issue has to do with using the entire Edamam uri as your resource ID, including the `http://www.edamam.com/ontologies/edamam.owl` part. There's all sorts of weird things about including a URL inside your URL - the simplest fix here is to chop off the first bit and just use the string of letters and numbers at the end of the uri. |
| Controller testing | You only have one test! Aside from success cases for all your routes, you should be testing what happens when things don't go well. For example, what if the user types in a search term with no results, or tries to find a recipe that doesn't exist? |
| **Styling** |  |
| Foundation Styling for responsive layout | yes |
| Search View | yes |
| List View | yes |
| Show View which opens in a new tab | yes |
| **API Features** | |
| The App attributes Edaman | yes |
| The VCR casettes do not contain the API key | no - your cassettes, which are checked into github for all the world to see, contain your edamam API keys! On a real site, this sort of thing would be a huge security problem! |
|  **Overall** |  |

Good work overall! The one thing that's missing is good controller testing. Being able to effectively test a controller is a culmination of all the Rails knowledge and understanding we've worked on so far, and it's a good learning experience. Make sure you practice it on Video Store.
